loadable_apps/loadable_sample/wifiapp/binary_update.c

```diff
@@ -148,25 +156,30 @@ static int binary_update_download_binary(binary_update_info_t *binary_info, int
 	ret = lseek(write_fd, 0, SEEK_SET);
 	if (ret != 0) {
 		printf("Failed to lseek %d, errno %d\n", ret, errno);
+		ret = ERROR;
 		goto errout_with_close_fd2;
 	}
 	ret = write(write_fd, (FAR uint8_t *)&crc_hash, CHECKSUM_SIZE);
 	if (ret != CHECKSUM_SIZE) {
 		printf("Failed to write %d\n", ret);
+		ret = ERROR;
 		goto errout_with_close_fd2;
 	}
 	printf("Download binary %s version %d Done!\n", APP_NAME, new_version);
 
-	return OK;
+	ret = OK;
 errout_with_close_fd2:
 	close(write_fd);
 errout_with_close_fd1:
 	close(read_fd);
 	if (ret < 0) {
 		fail_cnt++;
+		if (is_newfile_created) {
```

Comment:
```
if (ret != OK) {
   fail_cnt++;
   struct stat st; 
   if (stat(new_filepath, &st) == OK) {
       unlink(new_filepath);
   }
}
```

Reply:
But, with journaling, exist file is always valid.
If it stopped in the middle of downloading, written data in file is always valid.
so I think extra logic is needed to continue downloading.

Reply:
1. A file creation is verified when OK is returned by binary_manager_get_download_path. so it is enough to set and check a flag without calling stat to check file existing I think.
2. There is no logic for retrying write operation or something like what you said.
That's because retrying would be failed if the first try is failed.

Reply:
about #2, I can accept but you should use fs very restrictively, because we don't have a file permission, so every application can access that file.
So you need to unlink after stat, not flag. please consider delay of I/O and context switch.

Reply:
And If file is not exist, unlink will return -ENOENT, so if you want to try unlink for all cases, you can just call unlink without any checking.